### PR TITLE
Work on initial project layout

### PR DESCRIPTION
**Pull request by artem-v**

### CODE REVIEW

File: benchmarks/src/main/java/io/scalecube/benchmarks/BenchmarksState.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+import java.util.stream.LongStream;
+
+public class BenchmarksState<T extends BenchmarksState<T>> {
```
Comment (aharonha) [inline benchmarks/src/main/java/io/scalecube/benchmarks/BenchmarksState.java]: What is T?

File: benchmarks/src/main/java/io/scalecube/benchmarks/BenchmarksState.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  /**
+   * Runs given function in the state. It also executes {@link BenchmarksState#start()} before and
+   * {@link BenchmarksState#shutdown()} after.
+   * <p>
+   * NOTICE: It's only for synchronous code.
+   * </p>
+   *
+   * @param func function
+   */
+  public final void runForSync(Function<T, Function<Long, Object>> func) {
```
Comment (aharonha) [inline benchmarks/src/main/java/io/scalecube/benchmarks/BenchmarksState.java]: consder using `LongFunction<Object>`
please document what do you expect *this* function to return.

File: benchmarks/pom.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <groupId>io.projectreactor</groupId>
+            <artifactId>reactor-test</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard.metrics</groupId>
+            <artifactId>metrics-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-core</artifactId>
```
Comment (artem-v) [inline benchmarks/pom.xml]: Put plz `<scope>provided</scope>` and `<optional>true</optional>`

File: benchmarks/pom.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-generator-annprocess</artifactId>
+            <scope>provided</scope>
```
Comment (artem-v) [inline benchmarks/pom.xml]: Put plz `<scope>provided</scope>` and `<optional>true</optional>`

File: example/pom.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (artem-v) [inline example/pom.xml]: Rename plz module `example` to `examples` since we already have `benchmarks`; the plural.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <rules>
+                            <reactorModuleConvergence>
+                                <ignoreModuleDependencies>true</ignoreModuleDependencies>
+                            </reactorModuleConvergence>
+                            <dependencyConvergence/>
+                        </rules>
+                        <fail>false</fail>
```
Comment (artem-v) [inline pom.xml]: Plz dont disable enforcer plugin; remove `<fail>false</fail>` block

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+            <dependency>
+                <groupId>org.openjdk.jmh</groupId>
+                <artifactId>jmh-core</artifactId>
+                <version>${jmh.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.openjdk.jmh</groupId>
+                <artifactId>jmh-generator-annprocess</artifactId>
+                <version>${jmh.version}</version>
+                <scope>provided</scope>
```
Comment (artem-v) [inline pom.xml]: Add `<optional>true</optional>`

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+                <groupId>io.dropwizard.metrics</groupId>
+                <artifactId>metrics-core</artifactId>
+                <version>${metrics.version}</version>
+            </dependency>
+
+            <!-- Other dependencies -->
+            <dependency>
+                <groupId>org.openjdk.jmh</groupId>
+                <artifactId>jmh-core</artifactId>
+                <version>${jmh.version}</version>
```
Comment (artem-v) [inline pom.xml]: Add `<optional>true</optional>` and  `<scope>provided</scope>`

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+        <maven-shade-plugin.version>3.1.0</maven-shade-plugin.version>
+        <maven-enforcer-plugin.version>3.0.0-M2</maven-enforcer-plugin.version>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+
+            <dependency>
+                <groupId>io.projectreactor</groupId>
+                <artifactId>reactor-test</artifactId>
```
Comment (artem-v) [inline pom.xml]: Change on `reactor-core`

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+        </snapshotRepository>
+        <repository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
+
+    <modules>
+        <module>benchmarks</module>
+        <module>example</module>
```
Comment (artem-v) [inline pom.xml]: Plz rename module `examples` to `benchmarks-examples`; plz rename moudle `benchmarks` to `benchmarks-api`.

File: example/src/main/java/io/scalecube/benchmarks/example/ExampleServicesBenchmarksState.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+import io.scalecube.benchmarks.BenchmarksState;
+import io.scalecube.services.Microservices;
+import io.scalecube.services.ServiceCall;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+
+public class ExampleServicesBenchmarksState extends BenchmarksState<ExampleServicesBenchmarksState> {
```
Comment (artem-v) [inline example/src/main/java/io/scalecube/benchmarks/example/ExampleServicesBenchmarksState.java]: Remove dependency on `scalecube-services` ; just put a simplier examples w/o services and cluster.

File: pom.xml (hunk 7 above)
Comment (segabriel) [reply pom.xml]: done

File: pom.xml (hunk 8 above)
Comment (segabriel) [reply pom.xml]: done

**Pull request merged**